File: python/字符串/1165. 单行键盘.py
class Solution:
    def calculateTime(self, keyboard: str, word: str) -> int:
        d = dict()
        for i,v in enumerate(keyboard):
            d[v] = i
        tmp = keyboard[0] + word
        ans = 0
        pre = 0
        i = 1
        while i < len(tmp):
            t = d[tmp[i]]-pre
            ans += abs(t)
            pre = d[tmp[i]]
            i+=1
        return ans
s = Solution()
keyboard = "pqrstuvwxyzabcdefghijklmno"
word = "leetcode"
print(s.calculateTime(keyboard,word))

File: go/string/1165. 单行键盘.go
package string

func calculateTime(keyboard string, word string) int {
	var index = make(map[byte]int)
	for i := 0; i < len(keyboard); i++ {
		index[keyboard[i]] = i
	}
	tmp := string(keyboard[0]) + word

	ans := 0
	pre := 0
	for i := 1; i < len(tmp); i++ {
		t := index[tmp[i]] - pre
		if t < 0 {
			t = -t
		}
		ans += t
		pre = index[tmp[i]]
	}
	return ans
}
